### PR TITLE
fix(meeting): add collection super and move to /common

### PR DESCRIPTION
**Pull request by chrisadubois**
moved some stuff to /common and gave an example for how we could do collection extends with meeting info

I'm also considering moving constants.js to common directory, lmk what you guys think.

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import {MEETINGS} from '../constants';
+
+/**
+ * @class Collection
+ */
+export default class Collection {
+  namespace = MEETINGS;
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: `const`?

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  get(id) {
+    if (this[this.propertyName] && this[this.propertyName][id]) {
+      return this[this.propertyName][id];
+    }
+    return null;
+  }
+
+  /**
+   * @param {String} id the id of the meeting info instance to add to the collection
+   * @param {Any} set the thing to set in the collection
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: Probably name this something other than the function name, e.g., `somethingValue`.

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    return deleted;
+  }
+
+  /**
+   * @returns {Object} returns an object map of things stuffed into the collection
+   * @public
+   * @memberof Collection
+   */
+  getAll() {
+    return this.meetingInfos;
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: Will this always be `meetingInfos`? Or could it be generic?

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/collection.js [hunk 4]
```diff
@@ -1,41 +1,36 @@
...
-const MeetingInfoCollection = {
-  meetingInfos: {},
+import Collection from '../common/collection';
+import {MEETINGS} from '../constants';
+
+/**
+ * @class MeetingInfoCollection
+ */
+export default class MeetingInfoCollection extends Collection {
+  namespace = MEETINGS;
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/collection.js]: `const`?

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/collection.js [hunk 5]
```diff
@@ -1,41 +1,36 @@
...
+  }
 
   /**
-   *
    * @param {String} id ID of the meeting info you wish to retreive
    * @returns {MeetingInfo} returns a meeting info instance
+   * @public
+   * @memberof MeetingInfoCollection
    */
   get(id) {
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/collection.js]: Would you use the `super`'s `get` here, or do I misunderstand?

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js (hunk 1 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/common/collection.js]:
```
export default class Meetings extends SparkPlugin {
  namespace = MEETINGS;
```

this is how we do it everywhere

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js (hunk 3 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: oops, that should be this[this.propertyName]

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/collection.js (hunk 5 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/collection.js]: i am overriding the super's get here

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/collection.js (hunk 4 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/collection.js]: see above

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js (hunk 2 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: okay will change

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import {MEETINGS} from '../constants';
+
+/**
+ * @class Collection
+ */
+export default class Collection {
+  namespace = MEETINGS;
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: I don't think you need to namespace the class unless it extends one of the plugins.

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+   */
+  constructor(property) {
+    /**
+     * The property name for what collection will be stored, i.e., this.meetingInfos, this.meetings, etc.
+     * @instance
+     * @type {String}
+     * @public
+     * @memberof Meetings
+     */
+    this.propertyName = property;
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: Is it worth throwing an error if no property name is passed?

Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: good point. added.

File: packages/node_modules/@webex/plugin-meetings/src/common/collection.js (hunk 6 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/common/collection.js]: okay, we are doing it all over the place right now, so ill keep it and we can re-evaluate all of them at a later date.

**Comment by chrisadubois** [comment]
https://github.com/webex/spark-js-sdk/pull/1120

**Pull request closed without merging**